test/requests/links_scraper/genelinks.py

```diff
@@ -0,0 +1,136 @@
+import re
+import requests
+import urllib3
+import os
+import logging
+
+from urllib.request import urlopen as uReq
+from bs4 import BeautifulSoup as soup
+from urllib.parse import urljoin
+from urllib.parse import urlparse
+
+
+PORT = os.environ.get("PORT", "5004")
+TEMPLATE_PATH = "../wqflask/wqflask/templates"
+
+BROKEN_LINKS = set()
+
+
+def search_templates():
+    """searches for broken links in templates"""
+    html_parsed_pages = []
+    for subdir, dirs, files in os.walk(TEMPLATE_PATH):
+        for file in files:
+            file_path = os.path.join(subdir, file)
+            if file_path.endswith(".html"):
+                parsed_page = soup(
+                    open(file_path, encoding="utf8"), "html.parser")
+                html_parsed_pages.append(parsed_page)
+
+    return html_parsed_pages
+
+
+def is_valid_link(url_link):
+    try:
+        result = urlparse(url_link)
+        return all([result.scheme, result.netloc, result.path])
+    except Exception as e:
+        return False
+
+
+def test_link(link):
+    print(f'Checking -->{link}')
+    results = None
+    try:
+
+        results = requests.get(link, verify=False, timeout=10)
+        status_code = results.status_code
+
+    except Exception as e:
+        status_code = 408
+
+    if int(status_code) > 403:
+        return True
+
+    return False
```

Comment:
**nitpick:** Generally this should be one single return:
```
return int(status_code) > 403
```
:smile:

Reply:
Ahh, that makes a lot of sense and I'm pretty sure there are various places in GN2 where something like this could be done instead of having the if statement.